tests/fuzz/s2n_client_key_recv_fuzz_test.c

```diff
@@ -0,0 +1,174 @@
+    &s2n_ecdhe_rsa_with_3des_ede_cbc_sha,           /* 0xC0,0x12 */
+    &s2n_ecdhe_rsa_with_aes_128_cbc_sha,            /* 0xC0,0x13 */
+    &s2n_ecdhe_rsa_with_aes_256_cbc_sha,            /* 0xC0,0x14 */
+    &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       /* 0xC0,0x23 */
+    &s2n_ecdhe_ecdsa_with_aes_256_cbc_sha384,       /* 0xC0,0x24 */
+    &s2n_ecdhe_rsa_with_aes_128_cbc_sha256,         /* 0xC0,0x27 */
+    &s2n_ecdhe_rsa_with_aes_256_cbc_sha384,         /* 0xC0,0x28 */
+    &s2n_ecdhe_ecdsa_with_aes_128_gcm_sha256,       /* 0xC0,0x2B */
+    &s2n_ecdhe_ecdsa_with_aes_256_gcm_sha384,       /* 0xC0,0x2C */
+    &s2n_ecdhe_rsa_with_aes_128_gcm_sha256,         /* 0xC0,0x2F */
+    &s2n_ecdhe_rsa_with_aes_256_gcm_sha384,         /* 0xC0,0x30 */
+    &s2n_ecdhe_rsa_with_chacha20_poly1305_sha256,   /* 0xCC,0xA8 */
+    &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256, /* 0xCC,0xA9 */
+    &s2n_dhe_rsa_with_chacha20_poly1305_sha256,     /* 0xCC,0xAA */
+    &s2n_ecdhe_bike_rsa_with_aes_256_gcm_sha384,    /* 0xFF,0x04 */
+    &s2n_ecdhe_sike_rsa_with_aes_256_gcm_sha384,    /* 0xFF,0x08 */
+};
+
+static void s2n_fuzz_atexit()
+{
+    s2n_cleanup();
+    free(cert_chain_pem);
+    free(private_key_pem);
+    free(dhparams_pem);
+    s2n_config_free(config);
+    s2n_cert_chain_and_key_free(chain_and_key);
+}
+
+int LLVMFuzzerInitialize(const uint8_t *buf, size_t len)
+{
+#ifdef S2N_TEST_IN_FIPS_MODE
+    S2N_TEST_ENTER_FIPS_MODE();
+#endif
+
+    /* One time Diffie-Hellman negotiation to speed along fuzz tests*/
+    GUARD(s2n_init());
+    GUARD(atexit(s2n_fuzz_atexit));
+
+    cert_chain_pem = malloc(S2N_MAX_TEST_PEM_SIZE);
+    private_key_pem = malloc(S2N_MAX_TEST_PEM_SIZE);
+    dhparams_pem = malloc(S2N_MAX_TEST_PEM_SIZE);
+
+    notnull_check(cert_chain_pem);
+    notnull_check(private_key_pem);
+    notnull_check(dhparams_pem);
+
+    s2n_read_test_pem(S2N_DEFAULT_TEST_CERT_CHAIN, cert_chain_pem, S2N_MAX_TEST_PEM_SIZE);
+    s2n_read_test_pem(S2N_DEFAULT_TEST_PRIVATE_KEY, private_key_pem, S2N_MAX_TEST_PEM_SIZE);
+    s2n_read_test_pem(S2N_DEFAULT_TEST_DHPARAMS, dhparams_pem, S2N_MAX_TEST_PEM_SIZE);
+
+    config = s2n_config_new();
+    chain_and_key = s2n_cert_chain_and_key_new();
+
+    notnull_check(config);
+    notnull_check(chain_and_key);
+
+    s2n_cert_chain_and_key_load_pem(chain_and_key, cert_chain_pem, private_key_pem);
+    s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key);
+    s2n_config_add_dhparams(config, dhparams_pem);
+
+    cert = s2n_config_get_single_default_cert(config);
+    notnull_check(cert);
+
+    return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len)
+{
+    for(int version = 0; version < (sizeof(TLS_VERSIONS) / sizeof(TLS_VERSIONS[0])); version++)
+    {
+        for(int cipher = 0; cipher < (sizeof(s2n_all_cipher_suites) / sizeof(s2n_all_cipher_suites[0])); cipher++)
+        {
+            /* Setup */
+            if(s2n_all_cipher_suites[cipher]->key_exchange_alg != NULL)
```

Comment:
i'm assuming that this is needed for the tls 1.3 ciphers. perhaps use

```
if(s2n_all_cipher_suites[cipher]->key_exchange_alg == NULL) {
   continue;
}
```
instead.